contracts/modules/capital/Ramm.sol

```diff
@@ -417,6 +418,35 @@ contract Ramm is IRamm, MasterAwareV2 {
     );
   }
 
+  function getInitialObservations(
+    State memory initialState,
+    uint timestamp
+  ) public pure returns (Observation[3] memory initialObservations) {
+
+    Observation memory previousObservation;
+    uint endIdx = timestamp.divCeil(PERIOD_SIZE);
+
+    for (uint idx = endIdx - 2; idx <= endIdx; idx++) {
+      uint priceCumulativeAbove = previousObservation.priceCumulativeAbove;
+      uint priceCumulativeBelow = previousObservation.priceCumulativeBelow;
+
+      uint observationTimestamp = Math.min(timestamp, idx * PERIOD_SIZE);
```

Comment:
On the last iteration `timestamp` will be less than `idx * PERIOD_SIZE` so it will be used, but the `observationTimestamp` is not used when calculating the `timeElapsed`.